packages/editor/src/components/browser/role/AddRolePopover.tsx

```diff
@@ -0,0 +1,110 @@
+import { useState } from 'react';
+import {
+  BasicField,
+  Button,
+  Flex,
+  Input,
+  Message,
+  Popover,
+  PopoverArrow,
+  PopoverContent,
+  PopoverTrigger,
+  toast
+} from '@axonivy/ui-components';
+import { IvyIcons } from '@axonivy/ui-icons';
+import { useFunction } from '../../../context/useFunction';
+import { useQueryClient } from '@tanstack/react-query';
+import { useEditorContext } from '../../../context';
+import type { RoleMeta } from '@axonivy/inscription-protocol';
+import type { Table } from '@tanstack/react-table';
+import { useRoles } from '../../parts/common/responsible/useRoles';
+
+export const AddRolePopover = ({ value, table }: { value: string; table: Table<RoleMeta> }) => {
+  const [open, setOpen] = useState(false);
+  const { taskRoles } = useRoles();
+  const [newRoleName, setNewRoleName] = useState('');
+
+  const queryClient = useQueryClient();
+  const { context } = useEditorContext();
+  const addRole = useFunction(
+    'meta/workflow/addRole',
+    {
+      context,
+      newRole: { identifier: 'test', parent: 'Everybody' }
+    },
+    {
+      onSuccess: () => {
+        toast.info('Role successfully added');
+        queryClient.invalidateQueries({ queryKey: ['meta/workflow/roleTree', context] });
+        setOpen(false);
+      },
+      onError: error => {
+        toast.error('Failed to add new role', { description: error.message });
+      }
+    }
+  );
+
+  const newNameExists = (): boolean => {
+    return table.getRowModel().flatRows.some(row => row.original.id === newRoleName);
+  };
+
+  const newNameIsValid = (): boolean => {
+    if (newRoleName.length === 0) {
+      return false;
+    }
+    if (newNameExists()) {
+      return false;
+    }
+    const removedWhitespacesRoleName = newRoleName.replace(/\s+/g, '');
+    if (removedWhitespacesRoleName.length === 0) {
+      return false;
+    }
+    return true;
+  };
+
+  const isValidRowSelected = (): boolean => {
+    const selectedRow = table.getSelectedRowModel().flatRows[0]?.original;
+    if (!selectedRow) return false;
+
+    return !taskRoles.some(role => role.id === selectedRow.id);
+  };
+
+  return (
+    <Popover
+      open={open}
+      onOpenChange={e => {
+        setOpen(e);
+        if (e) setNewRoleName('');
+      }}
+    >
+      <PopoverTrigger asChild>
+        <Button icon={IvyIcons.Plus} aria-label={`Add Role to ${value}`} title={`Add Role to ${value}`} disabled={!isValidRowSelected()} />
+      </PopoverTrigger>
+      <PopoverContent sideOffset={12} collisionPadding={5} style={{ zIndex: '10000' }}>
+        <Flex direction='column' gap={2} alignItems='center'>
+          <BasicField label='New role name' style={{ width: '100%' }}>
+            <Input value={newRoleName} onChange={e => setNewRoleName(e.target.value)} style={{ zIndex: '1000' }} />
```

Comment:
why is this z-index needed? I understand the one on the content, but not sure about this one

Reply:
Yes, you are right. This is not needed 👍